### ISSUE TITLE
Semicolon at the end of an if-statement looks like a bug.

### ISSUE THREAD

**Issue by svigerske** [coin-or/Clp#72]
Issue created by migration from Trac.

Original creator: petter

Original creation time: 2015-07-26 12:56:11

Assignee: somebody

Version: 

The following code is from ClpSolver.cpp (starting at line 7262 in my revision).

```
		  for (int i = 0; i < numberRows2; i++) {
		    if (sub[iBlock].getRowStatus(i)==basic && fabs(solution[i])<1.0e-7)
		      solution[i]=0.0;
		    if (solution[i] > dualTolerance_) {
		      // at upper
		      if (saveUpper[i] > 1.0e20)
			solution[i]=0.0;
		      objValue += solution[i] * saveUpper[i];
		    } else if (solution[i] < -dualTolerance_) {
		      // at lower
		      if (saveLower[i] < -1.0e20);
			solution[i]=0.0;
		      objValue += solution[i] * saveLower[i];
		    } else {
		      solution[i]=0.0;
		    }
		  }
```

The semicolon at the end of the if-statement looks like a bug to me.

**Comment by svigerske** [comment]
Comment by ralacroix created at 2016-03-29 20:41:55

Fixed in 0ea94c5bb7 and released in CLP 1.16.10

**Comment by svigerske** [comment]
Comment by @jjhforrest created at 2016-03-30 08:46:47

Resolution: fixed
